code/datums/components/squeak.dm

```diff
@@ -99,17 +110,22 @@
 
 /datum/component/squeak/proc/on_equip(datum/source, mob/equipper, slot)
 	SIGNAL_HANDLER
-
 	holder = equipper
-	RegisterSignal(holder, COMSIG_MOVABLE_CROSSED, .proc/play_squeak_crossed)
 	RegisterSignal(holder, COMSIG_MOVABLE_DISPOSING, .proc/disposing_react, TRUE)
-	RegisterSignal(holder, COMSIG_PARENT_PREQDELETED, .proc/holder_deleted)
+	RegisterSignal(holder, COMSIG_PARENT_PREQDELETED, .proc/holder_deleted, TRUE)
+	//override for the preqdeleted is necessary because putting parent in hands sends the signal that this proc is registered towards,
+	//so putting an object in hands and then equipping the item on a clothing slot (without dropping it first)
+	//will always runtime without override = TRUE
+	AddElement(/datum/element/connect_loc, holder, holder_connections)
 
 /datum/component/squeak/proc/on_drop(datum/source, mob/user)
 	SIGNAL_HANDLER
-
-	UnregisterSignal(user, COMSIG_MOVABLE_CROSSED)
-	UnregisterSignal(user, COMSIG_MOVABLE_DISPOSING)
+	var/obj/parent_as_obj = parent
+	if(parent_as_obj.loc == holder)
+		return
+	RemoveElement(/datum/element/connect_loc, holder, holder_connections)
```

Comment:
@Mothblocks removed that second element added to holder since squeaky items activating when someone moves over holder hasnt worked literally since it was added https://github.com/tgstation/tgstation/commit/98128340544144d7f9669909feecdb1764b6b06f# (parent.loc will never be a turf if parent is equipped by a mob) and equipping a second squeaky item will 100% of the time runtime because connect_loc tries to register location_change to holder twice